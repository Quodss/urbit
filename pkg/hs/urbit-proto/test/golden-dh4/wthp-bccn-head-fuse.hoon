^-  [@ @]
=+
  ^-  $%  [%foo a=@ b=@]
          [%bar x=@]
          [%baz f=$-(@ @)]
      ==
  [%foo 1 2]
?-  +2.+2
  %foo  [a b]
  %bar  [x x]
  %baz  [(f 1) 2]
==
