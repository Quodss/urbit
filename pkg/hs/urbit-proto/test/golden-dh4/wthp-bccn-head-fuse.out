PARSED

^-  [@ @]
=+  ^-  $%  {$foo [a=@ b=@]}
            {$bar [x=@]}
            {$baz [f=$-(@ @)]}
        ==
    [%foo 1 2]
?-(+2.+2 %foo [a b], %bar [x x], %baz [(f 1) 2])

SOFT

^-  [@ @]
=+  ^-  $:  ?(%7303014 %7496034 %8020322)
            ^-  $
            ?:  ?=(%"foo" +2)
              [a=@ b=@]
            ?:  ?=(%"bar" +2)
              x=@
            ??  ?=(%"baz" +2)
            f=$-(@ @)
        ==
    [%foo 1 2]
?:  ?=(%"foo" +2.+2)
  [a b]
?:  ?=(%"bar" +2.+2)
  [x x]
??  ?=(%"baz" +2.+2)
[(f 1) 2]

HARD

^-  [@ @]
=+  ^-  $:  ?(%7303014 %7496034 %8020322)
            ^-  $
            ?:  ?=(%"foo" +2)
              [a=@ b=@]
            ?:  ?=(%"bar" +2)
              x=@
            ??  ?=(%"baz" +2)
            ^=  f
            $?  +7
            $|  [$;(@ *) *]
            ++  %  @
            --
        ==
    [%foo 1 2]
?:  ?=(%"foo" +2.+2)
  [a b]
?:  ?=(%"bar" +2.+2)
  [x x]
??  ?=(%"baz" +2.+2)
:-  =+  f
    %.+2(+6 1:+3)
2

CODE

=+  ["foo" 1 2]
?:  ?=(%"foo" +4)
  [+10 +11]
?:  ?=(%"bar" +4)
  [+5 +5]
[(+5 1) 2]

TYPE

[@ @]

BASE

[1 2]